lib/json3.js

```diff
@@ -231,61 +226,6 @@
       // Detect incomplete support for accessing string characters by index.
       var charIndexBuggy = has("bug-string-char-index");
 
-      if (has("bug-string-replace")) {
-        // Partial polyfill for buggy `String#replace` implementations.
-        stringReplace = function(pattern, fn) {
-          var value = this;
-          var lastIndex = pattern.lastIndex = 0, fullMatch, result = "";
-          while ((fullMatch = pattern.exec(value))) {
-            var index = fullMatch.index, match = fullMatch[0];
-            result += value.slice(lastIndex, index);
-            lastIndex = index + match.length;
-            result += fn(match);
-          }
-          if (lastIndex < value.length) {
-            result += value.slice(lastIndex);
-          }
-          return result;
-        };
-      }
-
-      // Internal: Determines if a property is a direct property of the given
-      // object. Delegates to the native `Object#hasOwnProperty` method.
-      var isProperty;
-      if (!(isProperty = objectProto.hasOwnProperty)) {
-        isProperty = function (property) {
-          var members = {}, constructor;
-          if ((members.__proto__ = null, members.__proto__ = {
-            // The *proto* property cannot be set multiple times in recent
-            // versions of Firefox and SeaMonkey.
-            "toString": 1
-          }, members).toString != getClass) {
-            // Safari <= 2.0.3 doesn't implement `Object#hasOwnProperty`, but
-            // supports the mutable *proto* property.
-            isProperty = function (property) {
-              // Capture and break the object's prototype chain (see section 8.6.2
-              // of the ES 5.1 spec). The parenthesized expression prevents an
-              // unsafe transformation by the Closure Compiler.
-              var original = this.__proto__, result = property in (this.__proto__ = null, this);
-              // Restore the original prototype chain.
-              this.__proto__ = original;
-              return result;
-            };
-          } else {
-            // Capture a reference to the top-level `Object` constructor.
```

Comment:
Are there any browsers that need the `hasOwnProperty` fix? Apart from that, it all seems fine.